SleepEngine/Engine/Components/HierarchyNode.h

```diff
@@ -5,11 +5,13 @@ BEGIN_NAMESPACE_SLEEP
 // allowes object add children, and updates them
 class HierarchyNode : public Component
 {
-    using Base = Component;
 public:
+    using Base = Component;
+    using ChildrenContainer = std::vector<std::unique_ptr<Object>>;
+
     HierarchyNode();
 
-    GET_SET(Object*, getParent, setParent, m_object)
+    GET_SET(Object*, getParent, setParent, m_parent)

```

Comment:
Also, please rename the getParent/setParent form Component class to getObject/setObject. As of now this shadows that declaration, which is very bad. My bad, forgot to rename that myself.

Reply:
Yeah, thanks. Done